### PR TITLE
Cache int

### PR DESCRIPTION
**Pull request by theramkrish8**

### CODE REVIEW

File: app/controller/groupController.go [hunk 1]
```diff
@@ -22,6 +25,10 @@ func AddNewGroup(ctx *gin.Context) {
 		return
 	}
 	res := service.AddNewGroup(&group)
+	if res.Status == enums.Status.SUCCESS {
```
Comment (Ram-srini) [inline app/controller/groupController.go]: Not required

File: app/controller/groupController.go [hunk 2]
```diff
@@ -35,6 +42,10 @@ func UpdateGroup(ctx *gin.Context) {
 		return
 	}
 	res := service.UpdateGroup(&group)
+	if res.Status == enums.Status.SUCCESS {
+		cachestore.SetCache(ctx, group.Code, group, common.CACHE_OBJ_LONG_TIMEOUT)
```
Comment (Ram-srini) [inline app/controller/groupController.go]: update only if exists

File: app/controller/groupController.go [hunk 3]
```diff
@@ -55,6 +66,12 @@ func ListAllGroups(ctx *gin.Context) {
...
 	val := ctx.Param("val")
-	res := service.FetchGroupByCodeOrID(key, val)
-	ctx.JSON(http.StatusOK, res)
+	if cachestore.CheckCache(ctx, val) {
+		res := cachestore.GetCache(ctx, val)
+		fmt.Println("Reading group details from cache!!")
+		ctx.JSON(http.StatusOK, res)
+	} else {
+		res := service.FetchGroupByCodeOrID(key, val)
+		ctx.JSON(http.StatusOK, res)
```
Comment (Ram-srini) [inline app/controller/groupController.go:78]: add to cache if cache miss

File: app/controller/schoolController.go [hunk 4]
```diff
@@ -21,6 +24,10 @@ func AddNewSchool(ctx *gin.Context) {
 		return
 	}
 	res := service.AddNewSchool(&school)
+	if res.Status == enums.Status.SUCCESS {
+		cachestore.SetCache(ctx, school.Code, school, common.CACHE_OBJ_LONG_TIMEOUT)
+		cachestore.SetCache(ctx, school.SchoolID.String(), school, common.CACHE_OBJ_LONG_TIMEOUT)
```
Comment (Ram-srini) [inline app/controller/schoolController.go]: Not required

File: app/controller/schoolController.go [hunk 5]
```diff
@@ -34,6 +41,10 @@ func UpdateSchoolProfile(ctx *gin.Context) {
 		return
 	}
 	res := service.UpdateSchool(&school)
+	if res.Status == enums.Status.SUCCESS {
+		cachestore.SetCache(ctx, school.Code, school, common.CACHE_OBJ_LONG_TIMEOUT)
```
Comment (Ram-srini) [inline app/controller/schoolController.go]: check if exists and check school obj have updated timestamp

File: app/controller/schoolController.go [hunk 6]
```diff
@@ -49,6 +60,12 @@ func ListAllSchools(ctx *gin.Context) {
...
 	val := ctx.Param("val")
-	res := service.FetchSchoolByCodeOrID(key, val)
-	ctx.JSON(http.StatusOK, res)
+	if cachestore.CheckCache(ctx, val) {
+		res := cachestore.GetCache(ctx, val)
+		fmt.Println("Reading school details from cache!!")
+		ctx.JSON(http.StatusOK, res)
+	} else {
+		res := service.FetchSchoolByCodeOrID(key, val)
+		ctx.JSON(http.StatusOK, res)
```
Comment (Ram-srini) [inline app/controller/schoolController.go:72]: add school object first time

File: app/controller/userController.go [hunk 7]
```diff
@@ -37,6 +42,10 @@ func UpdateUserProfile(ctx *gin.Context) {
 		return
 	}
 	res := service.UpdateUser(&user)
+	if res.Status == enums.Status.SUCCESS {
+		cachestore.SetCache(ctx, user.Mobile, user, common.CACHE_OBJ_LONG_TIMEOUT)
```
Comment (Ram-srini) [inline app/controller/userController.go]: update if exists and check user object last time

File: app/controller/userController.go [hunk 8]
```diff
@@ -81,6 +90,9 @@ func TransferRole(ctx *gin.Context) {
 		return
 	}
 	res := service.TransferUserRole(roleType, fromUserID, toUserID)
+	if res.Status == enums.Status.SUCCESS {
```
Comment (Ram-srini) [inline app/controller/userController.go:93]: if exists delete from cache with mobile no. user id

File: app/controller/userController.go [hunk 9]
```diff
@@ -102,6 +114,13 @@ func AddBulkUsers(ctx *gin.Context) {
 		return
 	}
 	res := service.AddBulkUser(users)
+	if res.Status == enums.Status.SUCCESS {
+		for i := 0; i < len(users); i++ {
+			u := users[i]
```
Comment (Ram-srini) [inline app/controller/userController.go]: Not required

File: app/controller/userController.go [hunk 10]
```diff
@@ -115,6 +134,13 @@ func UpdateBulkUsers(ctx *gin.Context) {
 		return
 	}
 	res := service.UpdateBulkUsers(users)
+	if res.Status == enums.Status.SUCCESS {
+		for i := 0; i < len(users); i++ {
+			u := users[i]
+			cachestore.SetCache(ctx, u.Mobile, u, common.CACHE_OBJ_LONG_TIMEOUT)
```
Comment (Ram-srini) [inline app/controller/userController.go]: Not reuired

File: app/controller/userController.go [hunk 11]
```diff
@@ -162,8 +191,15 @@ func ListUserByMobileOrID(ctx *gin.Context) {
...
 	}
-	res := service.FetchUserByMobileOrID(key, val)
-	ctx.JSON(http.StatusOK, res)
+	if cachestore.CheckCache(ctx, val) {
+		res := utility.GetSuccessResponse(cachestore.GetCache(ctx, val))
+		fmt.Println("Reading user details from cache!!")
+		ctx.JSON(http.StatusOK, res)
+	} else {
+		res := service.FetchUserByMobileOrID(key, val)
+		ctx.JSON(http.StatusOK, res)
```
Comment (Ram-srini) [inline app/controller/userController.go:158]: add first time to cache

File: app/repo/userRepo.go [hunk 12]
```diff
@@ -121,11 +121,11 @@ func (repo *userRepository) TransferRole(roleType enums.UserRole,
 
 	err := repo.coll.Find(orFilter).Select(cols).All(&users)
 	if err != nil {
-		return "", err
+		return "", nil, err
 	}
 	count := len(users)
 	if count < 2 {
-		return common.MSG_INSUFFICIENT_USER_COUNT + strconv.Itoa(count), nil
+		return common.MSG_INSUFFICIENT_USER_COUNT + strconv.Itoa(count), nil, err
```
Comment (lionheart92) [inline app/repo/userRepo.go]: return common.MSG_INSUFFICIENT_USER_COUNT + strconv.Itoa(count), nil, err should be changed to 
return common.MSG_INSUFFICIENT_USER_COUNT + strconv.Itoa(count), nil, nil

File: app/repo/userRepo.go [hunk 13]
```diff
@@ -150,9 +150,10 @@ func (repo *userRepository) TransferRole(roleType enums.UserRole,
 	}
 	_, errs := bulk.Run()
 	if errs != nil {
-		return "", err
+		return "", nil, err
```
Comment (lionheart92) [inline app/repo/userRepo.go]: it should be errs since the condition is "if **errs** != nil"

**Pull request merged**